extractor/src/test/java/org/schabi/newpipe/extractor/services/youtube/YoutubeSubscriberTest.java

```diff
@@ -0,0 +1,1180 @@
+/*
+package org.schabi.newpipe.extractor.services.youtube;
+
+import org.jsoup.nodes.Document;
+import org.jsoup.nodes.Element;
+import org.junit.BeforeClass;
+import org.junit.Test;
+import org.schabi.newpipe.DownloaderTestImpl;
+import org.schabi.newpipe.extractor.NewPipe;
+import org.schabi.newpipe.extractor.exceptions.ExtractionException;
+import org.schabi.newpipe.extractor.localization.Localization;
+import org.schabi.newpipe.extractor.services.youtube.extractors.YoutubeChannelExtractor;
+
+import java.io.IOException;
+
+import static org.junit.Assert.assertEquals;
+import static org.schabi.newpipe.extractor.ServiceList.YouTube;
+import static org.schabi.newpipe.extractor.utils.Utils.removeNumber;
+
+/**
+ * Test for {@link YoutubeChannelExtractor}
+ * and specifically YoutubeChannelExtractor.getSubscriberCount()
+ * in all the languages supported by YouTube.
+ * Takes a long time because we need to test make 146 requests to YouTube
+ * DON'T RUN ON MOBILE DATA
+ * <p>
+ * pattern for functions name:
+ * testlangcodeRegionabbreviation (Region is optional)
+ * eg:
+ * testenk = english thousand
+ * testfrCam = French (Canada) million
+ * testzhTwk = Chinese (Taiwan) thousand
+ */
+
+/*
+Commenting the whole file because otherwise it will slow down too much the CI test.
+And also, often one up to three tests fail if you launch the whole tests, because some requests fail
+(it could be reCAPTCHAs) but they, as of today (2020-02-16) success if you run each one.
+
+To run the test (to investigate maybe future problems),
+You should temporarily create « public Document getDoc() { return this.doc;} » method in YoutubeChannelExtractor.
+ */
+
+/*
+public class YoutubeSubscriberTest {
```

Comment:
> It takes every language and compares it to the english case (the default one), this way we can guarantee that the values parsed by the others are correct.

That's exactly what this file does.

> That way we can use less requests and test multiple things at once

Having one test for one thing is great for this situation, because you instantly know what failed with the pattern for function names I gave.
But I don't know what you did, maybe indeed it's better.

> Using the other approach, I already find some missing patterns here as well.

Can you share the file you used for the other approach?

> PS: Please don't commit a commented file. Consider the annotation Ignore passing the correct reason.
Thank you

Reply:
For example, yeah ``hi`` fails, but you have 29 tests each time, when I run the test it's not clear which one failed, or even if the whole failed.

Also, of course you can test more, because you test 29 videos with a request, because you request the feed. That makes 2320 test with only 80 request.
For the subscriber count, I'm using two request for each language.

Reply:
Hum, found how to test 112 times for a request.
https://www.youtube.com/feed/guide_builder

Reply:
Look at my last comment & commit, I refactored the test, it's should be better know.